maa-cli/src/config/task/client_type.rs

```diff
@@ -33,6 +35,37 @@ impl ClientType {
         }
     }
 
+    pub fn reset_time(&self) -> DateTime<Local> {
+        let timezone = self.timezone();
+        let server_reset_hour = 4;
+
+        let fixed_offset = FixedOffset::east_opt(timezone * 3600).unwrap();
+        let local_time = Utc::now()
+            .with_timezone(&fixed_offset)
+            .with_hour(server_reset_hour)
+            .unwrap()
+            .with_nanosecond(0)
+            .unwrap()
+            .with_minute(0)
+            .unwrap()
+            .with_second(0)
+            .unwrap();
+
+        debug!("Server {} reset time: {}", &self, local_time);
+        local_time.into()
+    }
+
+    pub fn timezone(self) -> i32 {
+        match self {
+            ClientType::Official => 8,  // utc+8
+            ClientType::Bilibili => 8,  // utc+8
+            ClientType::Txwy => 8,      // utc+8
+            ClientType::YoStarEN => -7, // utc-7
+            ClientType::YoStarJP => 9,  // utc+9
+            ClientType::YoStarKR => 9,  // utc+9
+        }
```

Comment:
This match can be simplified:
```suggestion
        use ClientType::*;
        match self {
            Official | Bilibili | Txwy => 8,
            YoStarEN => -7,
            YoStarJP, YoStarKR => 9,
        }
```